src/main/java/oyster/tasks/DeadlineTask.java

```diff
@@ -0,0 +1,85 @@
+package oyster.tasks;
+
+import java.time.LocalDateTime;
+import java.util.Scanner;
+
+import oyster.exceptions.TaskFieldException;
+import oyster.utils.DateTimeFormatter;
+
+/**
+ * DeadlineTask contains a deadline.
+ */
+public class DeadlineTask extends Task {
+    public static final String FILE_SYMBOL = "D";
+
+    private final LocalDateTime deadline;
+
+    /**
+     * Creates a DeadlineTask given its values.
+     *
+     * @param description The description of the Task.
+     * @param deadline The deadline of the Task.
+     */
+    public DeadlineTask(String description, LocalDateTime deadline) {
+        super(description);
+        this.deadline = deadline;
+    }
+
+    /**
+     * Creates a DeadlineTask given an input String to parse.
+     *
+     * @param input The string to parse into a DeadlineTask.
+     * @return DeadlineTask object.
+     */
+    public static DeadlineTask fromInput(String input) {
+        String name = "";
+        String deadline = "";
+        boolean isDeadlineNext = false;
```

Comment:
perhaps you could consider renaming this method to createFromInput, to signpost that this method creates a task. similarly for the other task types.